src/glue_launcher_lambda/glue_launcher.py

```diff
@@ -0,0 +1,517 @@
+
+        execute_athena_query(
+            args.manifest_s3_output_location, create_query, athena_client
+        )
+
+
+def execute_manifest_glue_job(
+    job_name,
+    cut_off_time_start,
+    cut_off_time_end,
+    margin_of_error,
+    snapshot_type,
+    import_type,
+    import_prefix,
+    export_prefix,
+    glue_client,
+):
+    """Executes the given job in aws glue.
+    Keyword arguments:
+    job_name -- the name of the job to execute
+    cut_off_time_start -- the time to not report any timestamps before (use None for all)
+    cut_off_time_end -- the time to not report any timestamps after
+    margin_of_error -- the margin of error time for the job
+    snapshot_type -- the type of snapshots the manifest were generated from (full or incremental)
+    import_type -- the type of import manifests with no space, i.e. "streaming_main" or "streaming_equality"
+    import_prefix -- the base s3 prefix for the import data
+    export_prefix -- the base s3 prefix for the export data
+    """
+    logger.info(f"Executing glue job with name of '{job_name}'")
+
+    import_prefix = import_prefix.rstrip("/")
+    export_prefix = export_prefix.rstrip("/")
+
+    cut_off_time_start_qualified = (
+        "0" if cut_off_time_start is None else str(cut_off_time_start)
+    )
+
+    logger.info(
+        f"Start time for job is '{cut_off_time_start_qualified}', import type is {import_type}, "
+        + f"snapshot type is {snapshot_type}, end time is '{cut_off_time_end}', "
+        + f"import prefix of '{import_prefix}', export prefix of '{export_prefix}' "
+        + f"and margin of error is '{margin_of_error}'"
+    )
+
+    job_run_start_result = glue_client.start_job_run(
+        JobName=job_name,
+        Arguments={
+            "--cut_off_time_start": cut_off_time_start_qualified,
+            "--cut_off_time_end": str(cut_off_time_end),
+            "--margin_of_error": str(margin_of_error),
+            "--import_type": import_type,
+            "--snapshot_type": snapshot_type,
+            "--import_prefix": import_prefix,
+            "--export_prefix": export_prefix,
+            "--enable-metrics": "",
+        },
+    )
+    job_run_id = job_run_start_result["JobRunId"]
+    logger.info(f"Glue job with name of {job_name} started run with id of {job_run_id}")
+
+
+def handler(event, context):
+    """Handle the event from AWS.
+    Args:
+    event (Object): The event details from AWS
+    context (Object): The context info from AWS
+    """
+    global args
+    global logger
+
+    args = get_parameters()
+    logger = setup_logging(args.log_level)
+
+    dumped_event = get_escaped_json_string(event)
+    logger.info(f'Event", "event": {dumped_event}, "mode": "handler')
+
+    detail_dict = get_and_validate_job_details(event)
+
+    job_name = detail_dict[JOB_NAME_KEY]
+    job_status = detail_dict[JOB_STATUS_KEY]
+    job_queue = detail_dict[JOB_QUEUE_KEY]
+
+    if job_status in IGNORED_JOB_STATUSES:
```

Comment:
This is vulnerable to another status being added by batch. I would prefer to do something like `if job_status not in FINISHED_JOB_STATUSES` or something.